src/docs/asciidoc/getting_started.adoc

```diff
@@ -153,6 +153,61 @@ HazelcastEnterpriseHD#2Nodes#1q2w3e4r5t
 ```
 
 
+[[running-in-modular-java]]
+=== Running in modular Java
+
+Java http://openjdk.java.net/projects/jigsaw/[project Jigsaw] brought a new Module System into Java 9 and newer. Hazelcast supports running in the modular environment. If you want to run your application with Hazelcast libraries on modulepath, use following module names:
+
+* `com.hazelcast.core` for `hazelcast-<version>.jar` and `hazelcast-enterprise-<version>.jar`
+* `com.hazelcast.client` for `hazelcast-client-<version>.jar` and `hazelcast-enterprise-client-<version>.jar`
+
+Don't use `hazelcast-all-<version>.jar` or `hazelcast-enterprise-all-<version>.jar` on modulepath as it could lead to problems in module dependencies for your application. You can still use them on the classpath.
+
+The Java Module System comes with stricter visibility rules. It affects Hazelcast which uses internal Java API to reach the best performance results.
+
+Hazelcast needs for its proper work `java.se` module and access to
+following Java packages:
```

Comment:
Hazelcast needs `java.se` module and access to the following Java packages for a proper work: